internal/ingress/controller/template/template.go

```diff
@@ -1217,27 +1228,28 @@ func buildOpentracing(c interface{}, s interface{}) string {
 
 	buf := bytes.NewBufferString("")
 
-	if cfg.DatadogCollectorHost != "" {
+	switch {
+	case cfg.DatadogCollectorHost != "":
```

Comment:
Besides being pretty weird and ugly, the previous code made sense. Each of collectors are exclusive, and if you try to set a config for Datadog and Zipkin, as an example, your nginx will fail to start because now you will have two `opentracing_load_tracer` directives.

So I think in this very specific case we should keep that weird `if...else`

Reply:
Reverted it